app/model/lib/search_queries.py

```diff
@@ -6,7 +6,7 @@
 
 
 def dynamical_query(all_advance_query):
-    base_query = "SELECT DISTINCT studyId"
+    base_query = "SELECT DISTINCT publicId"
```

Comment:
This caused a number of issues. The search really needs reworking. I would have liked to just adapt this to use ORM queries, but the logic is pretty tricky, so I'd have to write better tests first (the current ones only check for the SQL generation, they don't actually run the query).